### PR TITLE
feat: add requirements parser

### PR DESCRIPTION
**Pull request by ObserverOfTime**
Experimental because it's incomplete.

### CODE REVIEW

**Comment by clason** [comment]
Open question: Do we want to keep adding "experimental" parsers? I think it would be good to have some sort of objective minimal standard to enforce for parsers (which this one very well might meet).


Also, note that "experimental" is no longer a status in Nvim-treesitter 1.0; we only have "unsupported" (which implies no auto-updates). To me, "experimental" would imply an even lower status (that allows failing tests), which I'm not sure we want to have?

**Comment by ObserverOfTime** [comment]
The "experimental" status is currently rather arbitrary and I think it's fine to drop it for 1.0 and review the experimental parsers at that time.

But what would the minimal standard be? Supporting most language features and not breaking with every other change?

Our "worst" experimental parser I can think of is pug which not much worse than, say, bash in terms of features or c/cpp in terms of stability.

**Comment by clason** [comment]
> The "experimental" status is currently rather arbitrary and I think it's fine to drop it for 1.0 and review the experimental parsers at that time.

I agree; "experimental" seems to me like a cop-out to avoid maintenance expectations -- it's precisely that I would like to push back on. Either it's good enough to be used -- then it's no longer experimental -- or it's not -- and then it needs some more cooking time _before_ being added to nvim-treesitter. 

For 1.0, I have just translated "experimental" to "tier 4", but I'm happy to change that (or would welcome a PR to `main` that bumps the relevant parsers' tier from 4 to 3).

In either case, the expectation for adding a parser to nvim-treesitter is that it will be actively supported. Tier 4 is meant as a temporary status to avoid removing useful work.


So TL;DR: If you think this parser is ready to be used (even though there's still some gaps and rough edges), just drop the "experimental" annotation.

**Review by clason — APPROVED** [review]

File: lua/nvim-treesitter/parsers.lua [hunk 1]
```diff
@@ -1310,6 +1310,14 @@ list.rego = {
   maintainers = { "@FallenAngel97" },
 }
 
+list.requirements = {
+  install_info = {
+    url = "https://github.com/ObserverOfTime/tree-sitter-requirements",
+    files = { "src/parser.c" },
+  },
+  maintainers = { "@ObserverOfTime" },
```
Comment (clason) [inline lua/nvim-treesitter/parsers.lua:1318]: Maybe add a `readme_note` explaining what this is for? ("requirements" is _very_ generic...) Does this need to register a filetype?

Comment (ObserverOfTime) [reply lua/nvim-treesitter/parsers.lua:1318]: > Does this need to register a filetype?

I've submitted a PR to Vim.

Comment (ObserverOfTime) [reply lua/nvim-treesitter/parsers.lua:1318]: I wouldn't mind leaving this open until that's merged.

Comment (ObserverOfTime) [reply lua/nvim-treesitter/parsers.lua:1318]: It's ready now.
